rfc/rbac-ng.md

```diff
@@ -0,0 +1,154 @@
+A self-service API should learn from the equivalent feature in Clouddriver how to integrate with user secrets and manage account definitions to provide a similar API for Kayenta.
+Users will then be able to configure Kayenta accounts at runtime rather than requiring a Spinnaker admin to update Kayenta's configuration each time.
+Multi-tenant usage of Kayenta will become feasible.
+
+## Timeline
+
+The initial Kayenta and dependent work is being developed during 2023Q1.
+Further integration with other Spinnaker microservices, role policies, and permission change tracking would be targeted for afterwards in 2023.
+
+## Design
+
+This consists of a few related security elements: authentication, authorization, and credentials management.
+Authentication is handled by Gate typically, so this will reuse the existing `X-Spinnaker-User` header for denoting the principal of the request.
+User details are tracked by Fiat, so this data will need to be mirrored into OPA as a new backend.
+Authorization is the more complex part as permission data is stored by the various microservices that own the resources.
+These checks are typically handled through the `PermissionEvaluator` interface from Spring Security which takes three arguments: the authenticated user, the resource, and the permission.
+Permission evaluation will be given the context of what roles the requesting user has, what permissions apply to applications, and what permissions apply to accounts.
+As applications are checked by multiple services, these permissions will be pushed by Front50 into Fiat so that any service can use application permissions.
+External role data can either be pushed from Gate on login (and when service accounts are saved) or included as part of the input data to Fiat for policy decisions depending on if it's locally available (like being included as JWT metadata or X.509 attributes).
+The Fiat API will be updated to support permission checks on these new resources along with supporting role policies.
+
+Account permissions are most applicable to metric store accounts where only the `READ` and `WRITE` permissions apply.
+Granting read permission to a metric store account allows a user to fetch metrics by querying its `MetricsService` bean.
+Granting write permission to a metric store account allows a sid to manage permissions of the account, update the account configuration, and delete the account.
+Note that write permissions in this context are unrelated to writing metrics to a metrics store; metrics should be written to stores from the deployed applications being measured.
+Access controls will be defined in terms of role policies which are intended to replace direct use of permissions maps.
+
+Credentials management will be implemented through a SQL provider for managing metric store account definitions along with integration with user secrets.
```

Comment:
Is there scope to include SQL provider support for canary configs too in this RFC?

Reply:
Nevermind I just found that in future plans :)